heudiconv/cli/run.py

```diff
@@ -222,6 +222,10 @@ def get_parser():
                             help='Additional queue arguments passed as '
                             'single string of Argument=Value pairs space '
                             'separated.')
+    parser.add_argument('--skiptop', action='store_true',
+                        help='Skip creating of top-level bids files. '
+                        'Useful when running in batch mode to prevent '
+                        'possible race conditions.')
```

Comment:
our --help is already LONG enough and I expect that only more options would come to alter BIDS handling behavior.  What if we "convert" `--bids` option into something more useful?  
Especially since I think we are no longer using positional args, I think it should be easy to do:

<details>
<summary>Here is a prototype showing that we could use `args='*'` mode:</summary>

```shell
~/proj/repronim/reproin > cat /tmp/argparse_.py          
__version__ = '1'
from argparse import ArgumentParser
docstr = ("""Example:
         heudiconv -d rawdata/{subject} -o . -f heuristic.py -s s1 s2 s3""")
parser = ArgumentParser(description=docstr)
parser.add_argument('--version', action='version', version=__version__)

parser.add_argument('-b', '--bids', nargs='*',
                    choices=['notop', 'nostriptimes'],
                    help="""\
                    flag for output into BIDS structure.  Additional options which
                    could be specified:  notop - no top level metadata-files, 
                    nostriptimes - do not strip times from side-car files""")

out = parser.parse_args()
print(out)


~/proj/repronim/reproin > python /tmp/argparse_.py --help
usage: argparse_.py [-h] [--version]
                    [-b [{notop,nostriptimes} [{notop,nostriptimes} ...]]]

Example: heudiconv -d rawdata/{subject} -o . -f heuristic.py -s s1 s2 s3

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit
  -b [{notop,nostriptimes} [{notop,nostriptimes} ...]], --bids [{notop,nostriptimes} [{notop,nostriptimes} ...]]
                        flag for output into BIDS structure. Additional
                        options which could be specified: notop - no top level
                        metadata-files, nostriptimes - do not strip times from
                        side-car files

~/proj/repronim/reproin > python /tmp/argparse_.py       
Namespace(bids=None)                               

~/proj/repronim/reproin > python /tmp/argparse_.py -b
Namespace(bids=[])

~/proj/repronim/reproin > python /tmp/argparse_.py -b notop
Namespace(bids=['notop'])

~/proj/repronim/reproin > python /tmp/argparse_.py -b notop -b nostriptimes
Namespace(bids=['nostriptimes'])

~/proj/repronim/reproin > python /tmp/argparse_.py -b notop nostriptimes 
Namespace(bids=['notop', 'nostriptimes'])

~/proj/repronim/reproin > python /tmp/argparse_.py -b notop nostriptimes nonsense
usage: argparse_.py [-h] [--version]
                    [-b [{notop,nostriptimes} [{notop,nostriptimes} ...]]]
argparse_.py: error: argument -b/--bids: invalid choice: 'nonsense' (choose from 'notop', 'nostriptimes')

```
</details>

so we can discriminate when it is not specified (`None` -- current `False`) for when specified but without options (`[]` -- current True).  So, if instead of carrying around a bool `bids` we just carry `bids` a list of options with an empty list for `False`, `['no-options']` if no options were provided, and otherwise just a full list of options, we could later add other options and remain compatible while testing `if bids:` and check for options `if bids and 'notop' not in bids:` where needed.

Reply:
Ok I restructured it with your suggestion. Internally, I set args.bids to either True of False and add args.bids_options, but from the point of view of the user it works as you suggested.